### ISSUE TITLE
Same device for multiple customers in Community Edition

### ISSUE THREAD

**Issue by jortega11** [thingsboard/thingsboard#10407]
**Is your feature request related to a problem? Please describe.**
I would like to have the ability to assign one device to multiple customers and, as far as I know, this is not possible at the moment.

**Describe the solution you'd like**
It would be that simple, just add the ability to add devices, entities... to different customers.

**Describe alternatives you've considered**

**Additional context**

**Comment by ezeroldan24** [comment]
Hi jortega11, nice to meet you! What you can do is share the data between devices through the chain of rules. This would clone the data to 2 or more devices. I don't know if that's what you're looking for, but it makes sense that you can create just one, because it has a unique ID and can't be repeated.

**Comment by jortega11** [comment]
I don't think that would be a good solution. Yes, it can be a workaround, but the thing is that I have a physical device (i.e. a thermometer) and I have a bunch of different customers. 
How can I make them see that same device without duplicating any data?

**Comment by ezeroldan24** [comment]
To understand you better, do you want different customers to see the same device in a dashboard?

**Comment by jortega11** [comment]
Correct. There will be different customers that will be able to see one device or another. 

Some of them should be able to see one exact same device, but, as far as I know, you can only assign one device to one customer, so the other one will not be able to see it.

**Comment by devaskim** [comment]
@jortega11 Try to use entity views

**Comment by ezeroldan24** [comment]
Try the @devaskim  You won't mirror the device, but you will be able to see it on all clients.

**Comment by jortega11** [comment]
Okay, I get how can you do it using entity views.

I'll explain myself better. I have device A, B and C for Site 1 and device D and E for Site 2.

Until now I had a dashboard so you could click on the site you wanted and it updated with the new information (from devices A, B and C or devices D and E) depending if you clicked on Site 1 or 2.

Let's say that I have customer X and customer Y and I want to let Customer X see all devices but just devices A and E to Customer Y. My initial thought would be assigning the devices I wanted to the customers I wanted and that's it, but as we know, that is not possible. That is when entity views come in.

Seeing the devices in my dashboard works by clicking on a widget and then, with an action, stablishing the entity from the widget depending on the device you clicked on and then sending you to a new dashboard state with all the information of that device.
Can this behavior be replicated using entity views? Because with the test I have made so far it does not seem very possible or intuitive to do.

**Comment by devaskim** [comment]
> Can this behavior be replicated using entity views?

Maybe I something forgot about entity views, but I don't see any problems for your case.

**Comment by jortega11** [comment]
Okay, I could manage to get it working more or less the way I wanted. But I have to create an entity view for every combination of device and customer, if you have many devices and/or customers it will be such a pain. I think it is a bit dumb to do it that way instead of letting devices be duplicated on different customers, but at least it works.

I would like to request that feature because it will be way faster and comfortable to do it that way.

**Comment by devaskim** [comment]
> I would like to request that feature because it will be way faster and comfortable to do it that way.

Honestly I doubt it would be ever implemented, because of it requires redesign the core. But will see...

>  if you have many devices and/or customers it will be such a pain

It could be automated with rule chains and custom widgets.

**Comment by devaskim** [comment]
> I ask you something

who?)

To debug where is your data, put each input parameter of an action in `console.log()` function, then call this action, open browser console (i.e. DevTools) and look at the output.

In general, entity data is in `widgetContext.data`, but most or all widgets provide more convenient data container in one of action's input parameters.

**Comment by ezeroldan24** [comment]
I just noticed why it didn't show up, I hadn't loaded it into the widget data. Could you explain yourself better on that console?  I understand that pressing inspect on the web brings up the console. Will you have any images of a simple case to look at? It would be very helpful to me. Thank you! I'm a thingsboard developer, your supports were always very helpful.

**Comment by sudhirbits** [comment]
Reading through the comments, the use case appears to be one of multiple devices for the customer. Multiple users of the customer and having access to all the devices.

**Comment by jortega11** [comment]
No, it would be multiple customers, not users, having access to all devices.

> Reading through the comments, the use case appears to be one of multiple devices for the customer. Multiple users of the customer and having access to all the devices.

**Comment by mykhailokornieiev** [comment]
The feature request is rejected, since it is does not match the design of Thingsboard. Only one customer/tenant can be owner of the device.
